test/api/v3/integration/groups/POST-groups_groupId_join.test.js

```diff
@@ -203,6 +203,15 @@ describe('POST /group/:groupId/join', () => {
         await expect(invitedUser.get('/user')).to.eventually.have.nested.property('party._id', party._id);
       });
 
+      it('Issue #12291: accepting a redundant party invite will let the user stay in the party', async () => {
```

Comment:
so I tested this test in `develop` so without your fix but the test passes correctly meaning that either the bug doesn't exist or the test case isn't replicating it correctly

cc @SabreCat

Reply:
Ouch. I should have checked that!

The bug is replicable on the localhost, so I guess it does exist. The problem seems to be with the test case. I will look into it. Meanwhile, please let me know if you are able to identify the problem.

Reply:
@paglias I made a minor change in the update statement after looking at some of the code which used the update statement. And since we are making an update to the party attribute, I also added an additional check to verify if the update statement is working or not. 

Could you please check if this fails in the `develop` branch or not? 

```
        await invitedUser.update({
          'party._id': party._id,
        });
```

I would have checked it myself but I am not able to run the integration tests on my system. Getting these errors (Timeout because of `"before all"` and `"after all"`  - 
`[16:17:07] The following tasks did not complete: test:api-v3:integration`
`[16:17:07] Did you forget to signal async completion?`

While we are at it, do you know what's causing this? Or should I ask in the Aspiring Blacksmiths guild?

Reply:
it's working now @JalanshMunshi ! I hadn't notice the error with updating the user. Regarding the error you see when running the integration tests it might be that mongodb isn't running